src/NuGet.Clients/NuGet.VisualStudio.Common/ProjectSystems/ProjectNames.cs

```diff
@@ -45,36 +59,47 @@ public class ProjectNames : IEquatable<ProjectNames>
                 throw new ArgumentException(Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(customUniqueName));
             }
 
+            if (string.IsNullOrEmpty(projectId))
+            {
```

Comment:
In addition this check we better to add if `projectId` is actual guid. Also not equal to Guid.Empty which is default value, otherwise by some mistake we can get so many project with same Guid.Empty id.